File: Medium/1557. Minimum Number of Vertices to Reach All Nodes/Solution.py
class Solution(object):
    def findSmallestSetOfVertices(self, n, edges):
        ret = []
        inDegree = [0] * n

        for [u, v] in edges:
            inDegree[v] += 1

        for node in range(0, n):
            if not inDegree[node]:
                ret.append(node)

        return ret

File: Medium/1557. Minimum Number of Vertices to Reach All Nodes/Solution.cpp
class Solution {
public:
    vector<int> findSmallestSetOfVertices(int n, vector<vector<int>>& edges) {
        int inDegree[n];
        memset(inDegree, 0, sizeof(inDegree));
        for (auto &edge : edges) inDegree[edge[1]]++;
        vector<int> ret;

        for (int node = 0; node < n; node++)
            if(!inDegree[node]) ret.emplace_back(node);

        return ret;
    }
};